### DISCUSSION TITLE
Installation stuck on disk partitioning

### DISCUSSION THREAD

**Discussion by ladisnek** [elementary/installer#952] [Q&A] [answered]
Hi,

When I am trying to install Elementary OS I always get these errors when the disk partitioning starts.

ERROR: libdistinst: pointer in FFI is null
ERROR: errored while installing system: partition validation: the ESP partition must be at least 256MiB in size

I use the "Erase Disk and Install" choice. 

There was previously installed Windows. After first attempt I tried to delete partitions and partition the disk manually. Nothing has worked.
When I tried reinstalling Windows it works correctly. So I think there is problem with the Elementary installer. 

I tried installing several versions of Elementary including daily, RC, stable builds, but none of them works.

When I am in demo mode everything works correctly.

I have Lenovo Duet 3 11IAN8 tablet/laptop with Intel N200 CPU, and UFS SSD Kioxia drive.

Please help me, I loved Elementary on my old laptop and I would love to use it on this new too.

Thank you in advance for your answers!

EDIT 1: When tried installing clean Ubuntu, it seems there is problem with creating GRUB at the disk.

EDIT 2: certainly its problem with how the installer recognizes the partitions, see pictures:

EDIT 3: I browsed internet and I think it's issue with 4kb sector size of the disk, but no workaround has worked for me.

![20240324_140958](https://github.com/elementary/triage/assets/116750952/3518b239-1993-48fa-ab97-12706b4e4eb5)
![20240324_141015](https://github.com/elementary/triage/assets/116750952/28d501a3-ab43-4485-89ed-ec8d0254ce80)

**Comment by themanieldaniel** [comment]
I had a similar problem, I didn't delve into it very much, so I'm not sure if it is exactly the same, but I wanted a clean install so when the partition error came up I just went into advanced and deleted all partitions (even the boot partition), applied changes and then went back to click Erase and Install. It seemed to work fine after that. Maybe that will help?

**Reply by amarraff** [reply]
Experiencing this right now while trying to install Elementary on a 2015 Macbook Air. Drive is wiped, but installer keeps throwing error "the ESP partition must be at least 256 MiB in size" which is disappointing. I was looking forward to having Elementary running on a Mac.

**Reply by serversonic** [reply]
Same here on a early 2015 MacBook Air 7,1. I've been able to install Debian 13 with no problem to get EFI back on it, but the Elementary install fails at the same place. I can see that `/boot/efi` is 300MiB. There's a discussion from PopOS here starting in 2021 which may indicate the issue: https://github.com/pop-os/distinst/issues/239

**Comment by ryonakano** [comment, marked as answer]
Closing as a duplicate of https://github.com/elementary/installer/issues/448
